week0/pyomo_introduction.ipynb

```diff
@@ -0,0 +1,1074 @@
+    "\n",
+    "# Todo: create the list for y\n",
+    "y = [math.sin(xv) for xv in x]\n",
+    "\n",
+    "# Todo: Generate the figure\n",
+    "plt.plot(x, y)\n",
+    "plt.title(\"Trig: sin function\")\n",
+    "plt.xlabel(\"x in radians\")\n",
+    "plt.ylabel(\"sin(x)\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Importing and exporting data using Pandas\n",
+    "\n",
+    "Often, it is useful to output the data in a general format so it can be imported into other tools or presented in a familiar application. Python makes this easy with many great packages already available. The next code shows how to use the `pandas` package to create a dataframe and export the data to a csv file that we can import to excel. You could also consult [pandas documentation](https://pandas.pydata.org/pandas-docs/stable/) to see how to export the data directly to excel.\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Inline Exercise:</b>\n",
+    "Execute the code below that shows how to import some data into a DataFrame from the Pandas package and then export this data to a csv file.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "           x  sin(x) (radians)\n",
+      "0   0.000000      0.000000e+00\n",
+      "1   0.063467      6.342392e-02\n",
+      "2   0.126933      1.265925e-01\n",
+      "3   0.190400      1.892512e-01\n",
+      "4   0.253866      2.511480e-01\n",
+      "..       ...               ...\n",
+      "95  6.029319     -2.511480e-01\n",
+      "96  6.092786     -1.892512e-01\n",
+      "97  6.156252     -1.265925e-01\n",
+      "98  6.219719     -6.342392e-02\n",
+      "99  6.283185     -2.449294e-16\n",
+      "\n",
+      "[100 rows x 2 columns]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "df_sin = pd.DataFrame({\"x\": x, \"sin(x) (radians)\": y})\n",
+    "print(df_sin)\n",
+    "df_sin.to_csv(\"sin_data.csv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you go back to the browser tab that showed all the Jupyter notebook files and refresh, you will now see that there is a csv file with the x and y data. You can consult the Pandas documentation do learn about the many data analysis and statistical features of the `pandas` package.\n",
+    "\n",
+    "### Further Information\n",
+    "\n",
+    "Further information of the packages mentioned above can be found using the following links:\n",
+    "\n",
+    "* [numpy](https://numpy.org/devdocs/)\n",
+    "* [matplotlib](https://matplotlib.org/)\n",
+    "* [pandas](https://pandas.pydata.org/pandas-docs/stable/)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction to Pyomo\n",
```

Comment:
This Pyomo example is clear and easy for beginners. We might consider to have a combined Pyomo tutorial (from another Week 0 PR) if there is time.

Reply:
I mostly agree. Having two Pyomo introduction tutorials is a bit confusing. This is a nice simple example though and would lead nicely into my tutorial where you are adding other arguments to declare Vars.